### PR TITLE
proxy: remove sequence check

### LINKED ISSUE #409: Remove sequence check
## Bug Report

```
mysql
crw read2 65 4 0 [61 0 0 4 0 252 0 60 0 2 0 0 0 51 82 101 99 111 114 100 115 58 32 49 53 51 54 48 32 32 68 101 108 101 116 101 100 58 32 48 32 32 83 107 105 112 112 101 100 58 32 48 32 32 87 97 114 110 105 110 103 115 58 32 48]

tidb
 crw read2 72 4 65 [61 0 0 34 0 252 0 60 0 2 0 0 0 51 82 101 99 111 114 100 115 58 32 49 53 51 54 48 32 32 68 101 108 101 116 101 100 58 32 48 32 32 83 107 105 112 112 101 100 58 32 48 32 32 87 97 114 110 105 110 103 115 58 32 48]
```

tidb and mysql returns different uncompressed sequence. TiDB is correctly implemented as the mysql documentation. But mysql is implemented so and it wont change.

We can only choose one to validation. To be compatible with both, we could just remove these checks like jdbc.

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

Issue Number: close #409

Problem Summary: Remove sequence checks.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Notable changes

- [ ] Has configuration change
- [ ] Has HTTP API interfaces change
- [ ] Has tiproxyctl change
- [ ] Other user behavior changes

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/proxy/backend/authenticator.go [hunk 1]
```diff
@@ -214,7 +214,7 @@ loop:
 		if err != nil {
 			// tiproxy pp enabled, tidb pp disabled, tls disabled => invalid sequence
 			// tiproxy pp disabled, tidb pp enabled, tls disabled => invalid sequence
-			if pktIdx == 0 && errors.Is(err, pnet.ErrInvalidSequence) {
+			if pktIdx == 0 {
```
[the remark this answers is not in the document]
Comment (xhebox) [reply pkg/proxy/backend/authenticator.go:217]: A mysql error probably.

Or tidb will stuck, if it expects mysql packets but received a proxy header, which have a very large length interpreted as a mysql packet header.

File: pkg/proxy/backend/authenticator_test.go [hunk 2]
```diff
@@ -403,6 +403,10 @@ func TestProxyProtocol(t *testing.T) {
 	cfgOverriders := getCfgCombinations(cfgs)
 	for _, cfgs := range cfgOverriders {
 		ts, clean := newTestSuite(t, tc, cfgs...)
+		// invalid sequence detection removed, backend will stuck if clients insists to send proxy header.
```
Comment (djshow832) [inline pkg/proxy/backend/authenticator_test.go:406]: The test will stuck or the real client will stuck?

File: pkg/proxy/backend/authenticator.go [hunk 3]
```diff
@@ -235,6 +231,18 @@ loop:
...
+			for _, p := range []string{
+				"packets out of order",
+				"(nvali): d sequence",
+				"invalid sequence",
+				"PROXY Protocol",
+			} {
+				if strings.Contains(packetErr.Message, p) {
+					return errors.Wrap(ErrBackendPPV2, packetErr)
+				}
+			}
```
Comment (djshow832) [inline pkg/proxy/backend/authenticator.go:245]: I thought we could judge it by error ID. This error message may change anytime.

Comment (djshow832) [reply pkg/proxy/backend/authenticator.go:245]: If we can not judge the error ID, we can just report ErrBackendPPV2. I think in most cases it's ErrBackendPPV2 if reading the first packet fails.

Comment (xhebox) [reply pkg/proxy/backend/authenticator.go]: Done for 1156/8052, which may caused by other reasons, then return wrapped error.
If it contains "Proxy protocol" string, it returns unwrapped ErrBackendPPV2.

Comment (xhebox) [reply pkg/proxy/backend/authenticator.go]: Addressed comments. Reported the issue to TiDB.

**Review by djshow832 — APPROVED** [review]

**Pull request merged**